src/category_theory/reflect_isomorphisms.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.limits.limits
+import category_theory.limits.preserves
+
+open category_theory category_theory.limits
+
+namespace category_theory
+
+universes v₁ v₂ u₁ u₂
+
+variables {C : Type u₁} [𝒞 : category.{v₁} C]
+include 𝒞
+
+section reflects_iso
+variables {D : Type u₂} [𝒟 : category.{v₂} D]
+include 𝒟
+
+/--
+Define what it means for a functor `F : C ⥤ D` to reflect isomorphisms: for any
+morphism `f : A ⟶ B`, if `F.map f` is an isomorphism then `f` is as well.
+Note that we do not assume or require that `F` is faithful.
+-/
+class reflects_isomorphisms (F : C ⥤ D) :=
+(reflects : Π {A B : C} (f : A ⟶ B) [is_iso (F.map f)], is_iso f)
+
+-- Having this as an instance seems to break resolution, so let's not.
+/-- If `F` reflects isos and `F.map f` is an iso, then `f` is an iso. -/
+def is_iso_of_reflects_iso {A B : C} (f : A ⟶ B) (F : C ⥤ D)
+  [h : reflects_isomorphisms F] [is_iso (F.map f)] :
+  is_iso f :=
+reflects_isomorphisms.reflects F f
+
+end reflects_iso
+
+variables {J : Type v₁} [small_category J] {K : J ⥤ C}
+
+/--
+Given a cone morphism whose object part is an isomorphism, produce an
+isomorphism of cones.
+-/
+def cone_iso_of_hom_iso {K : J ⥤ C} {c d : cone K} (f : c ⟶ d) [i : is_iso f.hom] :
+  is_iso f :=
+{ inv :=
+  { hom := i.inv,
+    w' := λ j, (as_iso f.hom).inv_comp_eq.2 (f.w j).symm } }
+
+/--
+Given a cocone morphism whose object part is an isomorphism, produce an
+isomorphism of cocones.
+-/
+def cocone_iso_of_hom_iso {K : J ⥤ C} {c d : cocone K} (f : c ⟶ d) [i : is_iso f.hom] :
+  is_iso f :=
+{ inv :=
+  { hom := i.inv,
+    w' := λ j, (as_iso f.hom).comp_inv_eq.2 (f.w j).symm } }
+
+variables {D : Type u₂} [𝒟 : category.{v₁} D]
+include 𝒟
+
+-- TODO: should cones.functoriality take K as an explicit argument? I think so...
```

Comment:
I'd be happy to change this. I think I used to try to make too many arguments implicit.